### PR TITLE
new waitTx function

### LINKED ISSUE #1324: Refactor WaitTxToBeMined

### PR DESCRIPTION
**Pull request by ARR552**
Closes #1324

### What does this PR do?

Refactor waitTxToBeMined function

### Reviewers

- @ToniRamirezM 
- @cool-develope 
- @Mikelle

### CODE REVIEW

**Comment by cool-develope** [comment]
Could you explain more detail about the context? We don't need the revert reason further more?

File: test/operations/wait.go [hunk 1]
```diff
@@ -70,14 +70,20 @@ func Poll(interval, deadline time.Duration, condition ConditionFunc) error {
...
-	ctx := context.Background()
-	return Poll(DefaultInterval, timeout, func() (bool, error) {
-		return txMinedCondition(ctx, client, hash)
-	})
+func WaitTxToBeMined(parentCtx context.Context, client ethClienter, tx *types.Transaction, timeout time.Duration) error {
+	ctx, cancel := context.WithTimeout(parentCtx, timeout)
+	defer cancel()
+	_, err := bind.WaitMined(ctx, client, tx)
+	if err != nil {
+		log.Error("error waiting tx to be mined: ", err)
```
Comment (Mikelle) [inline test/operations/wait.go:82]:
```suggestion
		log.Errorf("error waiting tx %s to be mined: %w", tx.Hash(), err)
```

**Review by Mikelle — APPROVED** [review]

File: aggregator/aggregator.go [hunk 2]
```diff
@@ -148,7 +148,7 @@ func (a *Aggregator) tryToSendVerifiedBatch(ctx context.Context, ticker *time.Ti
 
 	if proof != nil && proof.Proof != nil {
 		log.Infof("sending verified proof to the ethereum smart contract, batchNumber %d", batchNumberToVerify)
-		a.EthTxManager.VerifyBatch(batchNumberToVerify, proof.Proof)
+		a.EthTxManager.VerifyBatch(ctx, batchNumberToVerify, proof.Proof)
```
Comment (ToniRamirezM) [inline aggregator/aggregator.go:151]: Can we make this function return an error? The aggregator should know what to do if an inner error happened

**Review by ToniRamirezM — CHANGES_REQUESTED** [review]

File: test/operations/wait.go (hunk 1 above)
Comment (cool-develope) [reply test/operations/wait.go]: Sometimes, `tx` can be `nil`, in that case, it will break the entire application

**Review by cool-develope — APPROVED** [review]

**Pull request merged**